src/test/java/org/mockitousage/constructor/CreatingMocksWithConstructorTest.java

```diff
@@ -70,7 +116,25 @@ public void exception_message_when_constructor_not_found() {
             fail();
         } catch (MockitoException e) {
             assertThat(e).hasMessage("Unable to create mock instance of type 'HasConstructor'");
-            assertThat(e.getCause()).hasMessageContaining("0-arg constructor");
+            assertThat(e.getCause()).hasMessageContaining("Please ensure that the target class has a 0-arg constructor.");
+        }
+    }
+
+    static class MultipleConstructors {
+        MultipleConstructors(Integer x) {}
+        MultipleConstructors(Number x) {}
+    }
+
+    @Test
+    public void exception_message_when_multiple_constructors_found() {
+        try {
+            //when
+            MultipleConstructors mock = mock(MultipleConstructors.class, withSettings().useConstructor(123).defaultAnswer(CALLS_REAL_METHODS));
```

Comment:
> 123 is integer so we should be able to find a matching constructor here, no?
But `Integer` is a subclass of `Number`, so both constructors would match. As per the discussion above, I think the safest approach would be to fail here.

In the first scenario you suggested, the only unambiguous call is `SomeClass(a)`. 
`SomeClass(b)` matches two constructors - `SomeClass(A)` and `SomeClass(B)`, and the same argument goes for `SomeClass(c)`. In fact, this is the same usecase as the `Integer` vs. `Number` above.

By the same reasoning, in the second scenario you suggest,  `SomeClass(c)` and `SomeClass(b)` are both ambiguous, but `SomeClass(a)` is not. I'll add a couple of tests to make this clearer. Again, this is the same usecase as `Integer` and `Number`.

In retrospect, this isn't a very clear test, as there's too much going on here - autoboxing and inheritence, all mixed together. I'll replace this test case with something more explicit to get the point across.